### PR TITLE
Added test code for ffs()

### LINKED ISSUE #1845: Add tests for ffs()
Implement some test cases for standard [ffs()](https://pubs.opengroup.org/onlinepubs/009695399/functions/ffs.html) routine

### PR DESCRIPTION
**Pull request by binkoni**
This fixes #1845
ffs() from compat/libc/include/strings.h is declared in the header,
However it is not defined anywhere.
So I made this as an inline function that calls bit_ffs() before adding
test code.

Signed-off-by: Byeonggon Lee <gonny952@gmail.com>

### CODE REVIEW

File: src/compat/libc/string/ffs.c [hunk 1]
```diff
@@ -0,0 +1,21 @@
+/**
+ * @file
+ * @brief Implementation of ffs() function.
+ *
+ * @date 20.03.15
+ * @author Byeonggon Lee
+ */
+
+int ffs(int i) {
```
Comment (alexkalmuk) [inline src/compat/libc/string/ffs.c]: Are you sure this implementation is required? You can search for a similar thing in embox.

Comment (alexkalmuk) [reply src/compat/libc/string/ffs.c]: I mean you can try to use existing bit functions.

Comment (alexkalmuk) [reply src/compat/libc/string/ffs.c]: Please do not squash the commits next time :) You can squash them right before merge.

Now I see that `bit_ffs` uses `unsigned long` as an argument. But `ffs` uses int.

Comment (alexkalmuk) [reply src/compat/libc/string/ffs.c]: So probably we cannot use `bit_ffs` here.

Comment (alexkalmuk) [reply src/compat/libc/string/ffs.c]: So we should check something like `ffs(-1)` in the tests too.

**Comment by alexkalmuk** [comment]
@binkoni Can I please ask you to add `ffs` implemention like it was before squash and also addtional tests for negative arguments?

**Comment by binkoni** [comment]
@alexkalmuk I also added negative arguments tests

**Comment by alexkalmuk** [comment]
> @alexkalmuk I also added negative arguments to the test and it passes the test
> 
> > Can I please ask you to add ffs implemention like it was before
> 
> Do you mean my removed ffs implementation?

Yes.

**Comment by binkoni** [comment]
@alexkalmuk 
```C
int ffs(int in) {
	for (int i = 0; i < sizeof(int) * 8; ++i)
		if (in & (1 << i))
			return i + 1;
	return 0;
}
```
My implementation was this. should I add it like before commit?

File: src/compat/libc/string/ffs.c (hunk 1 above)
Comment (binkoni) [reply src/compat/libc/string/ffs.c]: > Please do not squash the commits next time :) You can squash them right before merge
Okay I got it.

**Comment by alexkalmuk** [comment]
> @alexkalmuk
> 
> ```c
> int ffs(int in) {
> 	for (int i = 0; i < sizeof(int) * 8; ++i)
> 		if (in & (1 << i))
> 			return i + 1;
> 	return 0;
> }
> ```
> 
> My implementation was this. it also passes the test. should I add it again like previous(removed) commit?

Yes, please. By the way, it would be great to have two commits - one with implementation, and one with the test.

**Comment by binkoni** [comment]
I added my ffs() implementation too and it passes the test.
If you are fine, I'll squash the 3 commits into 2 commits.

File: src/compat/libc/include/strings.h [hunk 2]
```diff
@@ -13,6 +13,7 @@
 #include <defines/size_t.h>
 
 #include <sys/cdefs.h>
+#include <util/bit.h>
```
Comment (alexkalmuk) [inline src/compat/libc/include/strings.h]: Can you please remove this since it is not required anymore?

File: src/compat/libc/string/ffs.c [hunk 3]
```diff
@@ -0,0 +1,16 @@
+/**
+ * @file
+ * @brief Implementation of ffs() function.
+ *
+ * @date 20.03.15
```
Comment (alexkalmuk) [inline src/compat/libc/string/ffs.c]: date?

File: src/tests/stdlib/ffs_test.c [hunk 4]
```diff
@@ -0,0 +1,39 @@
+/**
+ * @file
+ * @brief Test of ffs() function.
+ *
+ * @date 20.03.15
```
Comment (alexkalmuk) [inline src/tests/stdlib/ffs_test.c]: date?

File: src/tests/stdlib/ffs_test.c [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+ * @author: Byeonggon Lee
```
Comment (alexkalmuk) [inline src/tests/stdlib/ffs_test.c]: remove `:` please.

File: src/compat/libc/string/ffs.c [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+ * @author Byeonggon Lee
+ */
+
+#include <strings.h>
+
+int ffs(int in) {
+	for (int i = 0; i < sizeof(int) * 8; ++i)
+		if (in & (1 << i))
+			return i + 1;
```
Comment (alexkalmuk) [inline src/compat/libc/string/ffs.c]: Can you please use braces? https://github.com/embox/embox/wiki/Code-Style#tabs-and-brackets

File: src/tests/stdlib/Mybuild [hunk 7]
```diff
@@ -82,4 +82,9 @@ module strchr_test {
 	depends embox.framework.LibFramework
 }
 
+module ffs_test {
+	source "ffs_test.c"
 
+	depends embox.compat.libc.str
```
Comment (alexkalmuk) [inline src/tests/stdlib/Mybuild]: Remove these dependencies please, we are trying to avoid test dependencies.

Comment (binkoni) [reply src/tests/stdlib/Mybuild]: Okay

**Comment by alexkalmuk** [comment]
@binkoni Great! Can you squash the changes into 2 commits and we can merge.

**Comment by binkoni** [comment]
Oh sorry, looks like I messed up other commits. I'll close this pull request and create new pull request
